### PR TITLE
feat(apply-context): Support MongoDB ObjectId as context variables

### PR DESCRIPTION
**Pull request by Hybrid-Force**
Add MongoDB ObjectId support as context variables, and hopefully other types of objects with `toJSON` methods returning strings.

### CODE REVIEW

File: src/utils/applyContext.ts [hunk 1]
```diff
@@ -36,7 +39,10 @@ export function applyContext<T extends object>(
     str.replace(reDelimiters, (_, path: string) => {
       const value = getValueFromPath(context, path);
       if (Array.isArray(value)) return `{${value}}`;
-      if (value instanceof Object) return 'undefined';
+      if (value instanceof Object) {
+        const json = value.toJSON?.();
```
Comment (roggervalf) [inline src/utils/applyContext.ts:43]: I'm seeing that the error is coming from this line because of the .

Comment (Hybrid-Force) [reply src/utils/applyContext.ts:43]: Got it. Optional chaining is not supported in Node 13 and below. I will fix it.

**Review by roggervalf — APPROVED** [review]
Thank you @Hybrid-Force, just for curiosity, how are you using this package?

**Comment by Hybrid-Force** [comment]
> Thank you @Hybrid-Force, just for curiosity, how are you using this package?

Thanks for this great lib! I am using it for API access control of a multi-tenant app that I am working on.

**Pull request merged**